### DISCUSSION TITLE
The first keyboard I tried....

### DISCUSSION THREAD

**Discussion by Braintoe** [florisboard/florisboard#1358] [Q&A]
... that might become dangerous to my beloved Smart Keyboard Pro :-) Iam looking forward to see this one evolve!

One question though: Does anyone know if it is possible to always show capitals on the buttons of the keyboard layout (i.e. turn off switching between lowercase and uppercase depending on how the state of the shift key)?

**Comment by patrickgold** [comment]
> ... that might become dangerous to my beloved Smart Keyboard Pro :-) Iam looking forward to see this one evolve!

Glad to hear that you like FlorisBoard despite the main issues that this project currently has (word suggestions, unresolved bugs, etc.). I am working really hard on improving things and adding new stuff (like the new preferences I am currently working on in the beta track).

>  One question though: Does anyone know if it is possible to always show capitals on the buttons of the keyboard layout (i.e. turn off switching between lowercase and uppercase depending on how the state of the shift key)?

A permanent caps state is not possible, but there's at some hacky way to do it: First enable "Remember caps-lock state" in the Floris Preferences (In the Typing tab). Second, on the keyboard double tap key so the shift key is orange. This is the caps lock mode. Third, never touch the shift key again and voila - you kinda have an always-capital state for the letters. Note that this state resets whenever the keyboard state resets (app reload, orientation change, system language change, etc.), but it is still something.

**Comment by Braintoe** [comment]
I do not care much about word suggestions and such, so I turned that off - and as far as I tried, the basic typing feature works flawlessly. Together with the ingenious clipboard history and the "swipe-over" feature for space and backspace this keyboard of yours is really nice :-)

From my point of view, these two are the core benefits of the keyboard that gave it immediately a place on my phone (but so far, I only played around with it this evening, so I might discover more ;-) ).

But you might have misunderstood my question: I do not want to actually type everything in caps - my question only related to the keyboard display. I never liked the Android way with the whole keyboard view changing depending on the state of the shift key. I would prefer to be able to turn this off and have a keyboard where the keys... well, look as the hardware ones on my PC keyboard: a capital letter shown on each button, regardless of the state of the shift key. I find this a lot easier for typing since the eye does not have to readjust to a different optic whenever the shift key is toggled. This (and the freely editable emoticon menu, see [my answer](https://github.com/florisboard/florisboard/issues/636) to the corresponding issue here) is what I love about Smart Keyboard Pro. If there is a chance these two feaures might be added here, I will most likely have one new keyboard on my phone. 

Edit - apart from typos above: 
Sadly, my programming skills are in a state where I would not force anyone to work with my code (CMD batch programming, enough C to get an Arduino do what I want - but the logic behind anything object-oriented is over my head, despite decade-long attempts to understand that longer than for an hour ¯\\\_(ツ)\_/¯ , so I could only offer to help with skins or icons as a thanks...
